### PR TITLE
source-wordpress contribution from bishalbera

### LINKED ISSUE #46087: [connector-request] WordPress
This is a **suggestion** to create a connector for **WordPress** using the UI Connector Builder.
The website for the connector is https://wordpress.org.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by bishalbera**
## What

This PR adds a new connector for source Wordpress (source-wordpress).

Contributor-provided connector description:

The WordPress connector  enables seamless data synchronization between your WordPress site and various destinations. With this connector, you can effortlessly extract and integrate blog posts, pages, comments, and other WordPress data into your preferred analytics or storage platform, enabling streamlined data analysis and reporting.
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `domain` | `string` | Domain. The domain of the WordPress site |  |
| `password` | `string` | Placeholder Password. Placeholder for basic HTTP auth password - should be set to empty string | x |
| `username` | `string` | Placeholder Username. Placeholder for basic HTTP auth username - should be set to empty string | x |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| users | id | DefaultPaginator | ✅ |  ❌  |
| posts | id | DefaultPaginator | ✅ |  ❌  |
| categories | id | DefaultPaginator | ✅ |  ❌  |
| plugins | plugin | DefaultPaginator | ✅ |  ❌  |
| blocks | id | DefaultPaginator | ✅ |  ❌  |
| comments | id | DefaultPaginator | ✅ |  ❌  |
| pages | id | DefaultPaginator | ✅ |  ❌  |
| tags | id | DefaultPaginator | ✅ |  ❌  |
| revisions | id | DefaultPaginator | ✅ |  ❌  |
| media | id | DefaultPaginator | ✅ |  ❌  |
| taxonomies | category | DefaultPaginator | ✅ |  ❌  |
| types |  | DefaultPaginator | ✅ |  ❌  |
| themes | stylesheet | DefaultPaginator | ✅ |  ❌  |
| statuses |  | DefaultPaginator | ✅ |  ❌  |
| search | id | DefaultPaginator | ✅ |  ❌  |
| settings |  | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

**Comment by bishalbera** [comment]
@marcosmarxm hey I want to add some lines regarding auth . Do i add that in the readme ?

**Review by natikgadzhi — DISMISSED** [review]
Don't merge before we triple-check tnis, @marcosmarxm. Sus.

**Comment by bishalbera** [comment]
> Don't merge before we triple-check tnis, @marcosmarxm. Sus.

hey @natikgadzhi I think I know why are you saying "Sus". Is it because in the config the domain value is not reachable? If so then its because I had to delete it for some SSL issues and switched to hostinger . But I'm also confused why is it showing that value while I've set [this](https://blanchedalmond-meerkat-467322.hostingersite.com/) in the ui builder . You can confirm it from the below screenshot also . 
Screenshot 2024-10-05 at 9 01 24 AM

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
The contribution is missing the connector doc? Did you made it using the UI Builder?

File: airbyte-integrations/connectors/source-wordpress/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+metadataSpecVersion: "1.0"
+data:
+  allowedHosts:
+    hosts:
+      - "https:"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-wordpress/metadata.yaml:5]: It doens't look good.

File: airbyte-integrations/connectors/source-wordpress/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,4772 @@
...
+      - username
+      - parent_id
+    properties:
+      domain:
+        type: string
+        description: The domain of the WordPress site
+        name: domain
+        order: 0
+        title: Domain
+        default: http://biplab.great-site.net
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-wordpress/manifest.yaml:552]: As far I know, Airbyte doesn't allow connections to `http` websites in Airbyte cloud. Maybe this is the problem you're facing.

File: airbyte-integrations/connectors/source-wordpress/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,4772 @@
...
+          string
+        name: username
+        order: 2
+        title: Placeholder Username
+        default: x
+        airbyte_secret: true
+      parent_id:
+        type: string
+        order: 3
+        title: Parent Id
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-wordpress/manifest.yaml:576]: What is Parent Id?

File: airbyte-integrations/connectors/source-wordpress/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,4772 @@
...
+            field_name: page
+            inject_into: request_parameter
+          pagination_strategy:
+            type: PageIncrement
+            page_size: 100
+            start_from_page: 1
+            inject_on_first_request: true
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: wp/v2/pages/{{ config['parent_id'] }}/revisions
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-wordpress/manifest.yaml:434]: This probably need to be a Parent Stream from page.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments. I'll sync with Natik about the domain.

**Comment by bishalbera** [comment]
> The contribution is missing the connector doc? Did you made it using the UI Builder?

ohh thats strange because I made it using the UI builder and all the rest of the pr that I made had it

File: airbyte-integrations/connectors/source-wordpress/manifest.yaml (hunk 2 above)
Comment (bishalbera) [reply airbyte-integrations/connectors/source-wordpress/manifest.yaml:552]: @marcosmarxm yes I also discovered that when I was using the mentioned domain here but I changed it to [this](https://blanchedalmond-meerkat-467322.hostingersite.com/) . I dont know why it isn't reflecting here though Its there in the UI builder. You can confirm that from the screenshot in the chat above.

File: airbyte-integrations/connectors/source-wordpress/manifest.yaml (hunk 3 above)
Comment (bishalbera) [reply airbyte-integrations/connectors/source-wordpress/manifest.yaml:576]: parent id refers to the id of the original page for which the revisions are being fetched.

File: airbyte-integrations/connectors/source-wordpress/manifest.yaml (hunk 4 above)
Comment (bishalbera) [reply airbyte-integrations/connectors/source-wordpress/manifest.yaml:434]: ohh do you mean setting the page stream as the parent stream and revisions as its child that way we can dynamically fetch the ids?

Comment (bishalbera) [reply airbyte-integrations/connectors/source-wordpress/manifest.yaml]: @marcosmarxm I've configured the pages stream to be used as parent stream for the revisions stream and removed the parent_id user input also by following [this](https://docs.airbyte.com/connector-development/connector-builder-ui/partitioning#parent-stream) guide.

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**